File: SimpleGallery/java/com/muheng/simplegallery/utils/BitmapUtils.java
package com.muheng.simplegallery.utils;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.drawable.BitmapDrawable;
import android.widget.ImageView;

public class BitmapUtils {
    public static void setImageViewBitmap(ImageView view, Bitmap bitmap) {
        if (view != null && bitmap != null) {
            BitmapDrawable bd = (BitmapDrawable) view.getDrawable();
            if (bd != null) {
                bd.getBitmap().recycle();
                view.setImageBitmap(null);
            }
            view.setImageBitmap(bitmap);
        }
    }
    public static Bitmap downSamplingBitmap(String filePath, int inSampleSize) {
        Bitmap bitmap = null;
        try {
            BitmapFactory.Options options = new BitmapFactory.Options();
            options.inSampleSize = inSampleSize;
            bitmap = BitmapFactory.decodeFile(filePath, options);
        } catch(Exception e) {
            e.printStackTrace();
        }
        return bitmap;
    }
}

File: PhotoViewer/java/com/muheng/photoviewer/utils/BitmapUtils.kt
package com.muheng.photoviewer.utils

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.net.Uri
import java.io.File

class BitmapUtils {
    companion object {
        val TAG = "BitmapUtils"

        @Throws(Exception::class)
        fun decodeImage(uri : Uri, smapleSize : Int) : Bitmap {
            var options = BitmapFactory.Options()
            options.inSampleSize = smapleSize
            return BitmapFactory.decodeFile(File(uri.toString()).path, options)
        }
    }
}